addons/hr_expense/models/hr_expense.py

```diff
@@ -0,0 +1,320 @@
+            # post the journal entry if 'Skip 'Draft' State for Manual Entries' is checked
+            move.write({'line_ids': lines})
+            exp.write({'account_move_id': move.id, 'state': 'post'})
+            if exp.payment_mode == 'company_account':
+                exp.paid_expenses()
+        move.post()
+        return True
+
+    @api.multi
+    def _move_line_get(self):
+        account_move = []
+        Tax = self.env['account.tax']
+        for expense in self:
+            if expense.product_id:
+                account = expense.product_id.property_account_expense_id
+            if not account:
+                account = expense.product_id.categ_id.property_account_expense_categ_id
+            if not account:
+                raise UserError(_("No Expense account found for the product %s (or for it's category), please configure one.") % (expense.product_id.name))
+            else:
+                account = self.env['ir.property'].with_context(force_company=expense.company_id.id).get('property_account_expense_categ_id', 'product.category')
+                if not account:
+                    raise UserError(_('Please configure Default Expense account for Product purchase: `property_account_expense_categ_id`.'))
+            move_line = {
+                    'type': 'src',
+                    'name': expense.name.split('\n')[0][:64],
+                    'price_unit': expense.unit_amount,
+                    'quantity': expense.quantity,
+                    'price': expense.total_amount,
+                    'account_id': account.id,
+                    'product_id': expense.product_id.id,
+                    'uos_id': expense.product_uom_id.id,
+                    'account_analytic_id': expense.analytic_account_id.id,
+                }
+            account_move.append(move_line)
+
+            # Calculate tax lines and adjust base line
+            taxes = expense.tax_ids.compute_all(expense.unit_amount, expense.currency_id, expense.quantity, expense.product_id)
+            account_move[-1]['price'] = taxes['total_excluded']
+            account_move[-1]['tax_ids'] = expense.tax_ids.id
+            for tax in taxes['taxes']:
+                account_move.append({
+                    'type': 'tax',
+                    'name': tax['name'],
+                    'price_unit': tax['amount'],
+                    'quantity': 1,
+                    'price': tax['amount'],
+                    'account_id': tax['account_id'] or move_line['account_id'],
+                    'tax_line_id': tax['id'],
+                })
+        return account_move
+
+    @api.multi
+    def action_view_move(self):
+        '''
+        This function returns an action that display existing account.move of given expense ids.
+        '''
+        self.ensure_one()
+        try:
+            view_id = self.env.ref('account.view_move_form').id
+        except ValueError as e:
+            view_id = False
```

Comment:
You can avoid "try...except" by passing "raise_if_not_found=False" param to ref(...)
view_ref = self.env.ref('account.view_move_form', raise_if_not_found=False)
...
...
'view_id': view_ref.id if view_ref else False,